solr/modules/opentelemetry/src/java/org/apache/solr/opentelemetry/OtelTracerConfigurator.java

```diff
@@ -65,7 +65,9 @@ void prepareConfiguration(NamedList<?> args) {
     setDefaultIfNotConfigured("OTEL_EXPORTER_OTLP_PROTOCOL", "grpc");
     setDefaultIfNotConfigured("OTEL_TRACES_SAMPLER", "parentbased_always_on");
     setDefaultIfNotConfigured("OTEL_PROPAGATORS", "tracecontext,baggage");
-    addOtelResourceAttributes(Map.of("host.name", System.getProperty("host")));
+    if (System.getProperty("host") != null) {
```

Comment:
I was surprised too to learn that SOLR_HOST is not always set in our start script. And also learnt that `Map.of(k,v)` does null-check both for key and value. Those two facts together caused this. Although only likely a problem in dev environments, it's nice to fix.

Reply:
> And also learnt that Map.of(k,v) does null-check both for key and value.

yeah, this one bit me a few times too. almost all of the immutable collection utils (like List.of) do not allow nulls. weird rule.

> Although only likely a problem in dev environments, it's nice to fix.

agreed. although my preference would have been a default instead of missing value.